### ISSUE TITLE
Interesting NPM package

### ISSUE THREAD

**Issue by lpatiny** [cheminfo/xrd#1]
https://www.npmjs.com/package/xrd-calculate

**Comment by kjappelbaum** [comment]
it might be worth to also look into cif visualizers again. Tilde lab has a nice blog post https://blog.tilde.pro/in-browser-plugin-free-cif-visualization-comparison-of-open-source-engines-a3d0b4098660

**Comment by lpatiny** [comment]
2 are GPL licence and not compatible with the MIT licence of the visualizer.

Is
![image](https://user-images.githubusercontent.com/1484241/84622725-00589700-aede-11ea-8b9f-da5a4849b279.png)
Good enough for your needs ?

**Comment by kjappelbaum** [comment]
Yes, and I did not want to suggest that it should be changed, but it might be worth trying if this is a bit neater/faster (as they claim).

**Comment by kjappelbaum** [comment]
I had a look at the `xrd-calculate` ... seems they simply transpiled some C code with Emscripten ... but it seems to do what we need. It's a pitty that they only export the function to calculate the pattern based on a file (and not on the file data)

**Comment by lpatiny** [comment]
Web assembly is definitely an option if you have access to the C code

**Comment by kjappelbaum** [comment]
Anyhow, I'm using it now and put the prediction into another analysis object ... https://github.com/kjappelbaum/xrd-analysis/commit/2aca34b2783584ff1f468adbfb559b2267b9ae1c but I need to test a bit if the predictions are indeed correct ..

**Comment by lpatiny** [comment]
We should just not forget that xrd-analysis has to be build for the browser and not only for node.
